python/scope.py

```diff
@@ -1036,10 +1048,30 @@ async def eval_command(self, tree):
 
                 scope.assign_to_pattern(pattern, i, certain=True)
                 exit, value = await scope.eval_command(code)
-                if exit:
+                if exit == True: # exit can be a string
                     return True, value
+                if exit == "continue":
+                    continue
+        elif command.data == "while":
+            var, code = command.children
+            val = await self.eval_expr(var)
+            while val:
+                scope = self.new_scope()
+
+
+                exit, value = await scope.eval_command(code)
+                if exit == True: # exit can be a string
+                    return True, value
```

Comment:
Hmm, wouldn't this make `break` return out of the function? Also, you can avoid the comparison with `True` by moving this after `if exit == "continue":`